indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -125,10 +124,13 @@ public void run()
         log.info("Started WorkerTaskMonitor.");
         started = true;
       }
+      catch (InterruptedException e) {
+        throw e;
```

Comment:
The need of re-setting of the interrupt flag depends on what happens in the caller.

If the caller catches `Exception`, then logs and keeps going, then the behavior is potentially going to violate expectations because the *NEXT* task in the loop might also stall and need interrupting.

So the question is, do you expect an interrupt to mean "quit now" or do you expect an interrupt to mean "quit the explicit task you are doing because it is stuck, but potentially keep going"?